controllers/trench_controller.go

```diff
@@ -68,109 +66,28 @@ func (r *TrenchReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
+		if apierrors.IsNotFound(err) {
 			return reconcile.Result{}, nil
 		}
 		return ctrl.Result{}, err
 	}
 
-	// Check if the configmap already exists, if not create a new one
-	configMap := &corev1.ConfigMap{}
-	err = r.Client.Get(ctx, client.ObjectKey{Namespace: trench.Name, Name: trench.Spec.ConfigMapName}, configMap)
-	if apierrors.IsNotFound(err) {
-		configMap = r.buildConfigMap(trench)
-		err := r.Client.Create(ctx, configMap)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-		return ctrl.Result{}, nil
-	}
+	executor := reconciler.NewExecutor(r.Scheme, r.Client, ctx, trench)
+	meridio := reconciler.Meridio{}
+	err = meridio.ReconcileAll(executor, trench)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 
-	// Ensure the configmap contains the same data as the spec
-	if !configMapValid(trench, configMap) {
-		r.setConfigMapData(trench, configMap)
-		err = r.Update(ctx, configMap)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-		return ctrl.Result{}, nil
-	}
-
 	return ctrl.Result{}, nil
 }
 
-func (r *TrenchReconciler) setConfigMapData(trench *meridiov1alpha1.Trench, configMap *corev1.ConfigMap) {
-	config := &Config{
-		VIPs: trench.Spec.VIPs,
-	}
-	configYAML, err := yaml.Marshal(&config)
-	if err != nil {
-		log.Fatalf("error yaml.Marshal: %v", err)
-	}
-	configMap.Data = map[string]string{
-		MeridioConfigKey: string(configYAML),
-	}
-	configMap.Data[MeridioConfigKey] = string(configYAML)
-}
-
-func (r *TrenchReconciler) buildConfigMap(trench *meridiov1alpha1.Trench) *corev1.ConfigMap {
-	configMap := &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      trench.Spec.ConfigMapName,
-			Namespace: trench.Name,
-		},
-	}
-	r.setConfigMapData(trench, configMap)
-
-	controllerutil.SetControllerReference(trench, configMap, r.Scheme)
-	return configMap
-}
-
-func configMapValid(trench *meridiov1alpha1.Trench, configMap *corev1.ConfigMap) bool {
-	configuration, ok := configMap.Data[MeridioConfigKey]
-	if !ok {
-		return false
-	}
-	config := &Config{}
-	err := yaml.Unmarshal([]byte(configuration), &config)
-	if err != nil {
-		return false
-	}
-	if !vipListsEqual(config.VIPs, trench.Spec.VIPs) {
-		return false
-	}
-	return true
-}
-
-func vipListsEqual(vipListA []string, vipListB []string) bool {
-	vipsA := make(map[string]struct{})
-	vipsB := make(map[string]struct{})
-	for _, vip := range vipListA {
-		vipsA[vip] = struct{}{}
-	}
-	for _, vip := range vipListB {
-		vipsB[vip] = struct{}{}
-	}
-	for _, vip := range vipListA {
-		if _, ok := vipsB[vip]; !ok {
-			return false
-		}
-	}
-	for _, vip := range vipListB {
-		if _, ok := vipsA[vip]; !ok {
-			return false
-		}
-	}
-	return true
-}
-
 // SetupWithManager sets up the controller with the Manager.
 func (r *TrenchReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&meridiov1alpha1.Trench{}).
 		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.Service{}).
+		Owns(&appsv1.Deployment{}).
 		Complete(r)
```

Comment:
More resources are deployed by the operator (Role, Rolebinding, Service account...), should they be added here?

Reply:
All resouces should be added now